async_tcp_server.cpp

```diff
@@ -0,0 +1,101 @@
+#include "async_tcp_server.hpp"
+#include <algorithm>
+#include <cstring>
+#include <iostream>
+#include <sys/select.h>
+
+AsyncTCPServer::AsyncTCPServer() {}
+
+AsyncTCPServer::AsyncTCPServer(Socket_sptr master_socket, fd_set fdsetRead): m_fdsetRead(fdsetRead)
+{
+    FD_ZERO(&m_fdsetRead);
+    FD_SET(master_socket->GetSocketHandle(), &m_fdsetRead);
+    m_aSockets.push_back(master_socket);
+}
+
+Socket_sptr AsyncTCPServer::GetSocketByFD(int fd) const
+{
+    for(auto socket : m_aSockets)
```

Comment:
Lets try to use standard algorithms instead of own implementation (BTW, `for` loop here will copy each vector element just to compare it with `fd`).
Function you're looking for - `std::find`. But since we have no `operator ==` in Socket class I guess it should not work. :)
For your practice, if you're not familiar with C++ `lambda`s, try also to use `std::find_if`, and pass lambda as predicat. Just as exercise ;)

Reply:
I just understood i was wrong about `std::find` will not be working - it will, 'cause `Socket` is POD class. But common comparision will compare all its fields, while we really need to compare only FD inside it.